### DISCUSSION TITLE
Thermal Hydraulics tutorial issues

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#22240] [Q&A General] [answered]
Hello,

I am working through the thermal hydraulics module tutorial and the base input file will not compile for the conjugate heat transfer step located here: https://mooseframework.inl.gov/bison/modules/thermal_hydraulics/tutorials/single_phase_flow/step02.html

I am trying to run the base tutorial before modifying the file, but both my modified code and the original input file are throwing the same error shown in the picture below.
<img width="270" alt="Screen Shot 2022-09-27 at 3 05 45 PM" src="https://user-images.githubusercontent.com/86804307/192614099-fe2562da-2644-44fe-b89a-4179748a998b.png">

*** ERROR ***
Unable to find closest node!

I am not sure as to how the code is struggling to find the closest node since the only thing that has changed from step one to step 2 is adding the heat structure values which have the same geometry and elements as the core structure. Any tips would be helpful. Thanks.

UPDATE: I only get this error when using MPI. The code compiles and converges when only using a single processor.

**Comment by GiudGiud** [comment]
@joshuahansel

**Comment by joshuahansel** [comment, marked as answer]
Are you running with a distributed mesh? I'm unable to reproduce your error (I ran with MPI). However, if I additionally use a distributed mesh, I just get a segmentation fault...

I'm aware of one of our THM components, `HeatTransferFromHeatStructure1Phase` (which is indeed in this model), not being compatible with distributed meshes. I see from my run that it doesn't give an error message, so I'll at least fix that.

Just curious: are you planning on a model where you actually need to use MPI? Not that it's not possible, but usually THM models have a relatively small system size due to being primarily 1D, and thus it's not terribly beneficial to use more processors. In multi-app setups where we have a large number of flow channels, we usually restrict each THM sub-app to 1 processor. This is not to say that what you're trying to do is wrong, just that I'd like to be aware of any need to prioritize distributed mesh support in THM in the future.
